xmake/actions/run/main.lua

```diff
@@ -44,23 +44,18 @@ function _do_run_target(target)
     -- get the absolute target file path
     local targetfile = path.absolute(target:targetfile())
 
-    -- add run environments
-    local addrunenvs, setrunenvs = make_runenvs(target)
-    for name, values in pairs(addrunenvs) do
-        os.addenv(name, table.unpack(table.wrap(values)))
-    end
-    for name, value in pairs(setrunenvs) do
-        os.setenv(name, table.unpack(table.wrap(value)))
-    end
+    -- build run environments
+    local addrunenvs, setrunenvs = runenvs.make(target)
 
     -- get run arguments
     local args = table.wrap(option.get("arguments") or target:get("runargs"))
 
     -- debugging?
     if option.get("debug") then
-        debugger.run(targetfile, args, {curdir = rundir})
+        debugger.run(targetfile, args, {curdir = rundir, addrunenvs = addrunenvs, setrunenvs = setrunenvs})
```

Comment:
please pass `{envs = ..}`

Reply:
it's on purpose so the renderdoc debugger can know which envs are to replace and which are to add